### ISSUE TITLE
nrf: implement checking for pin in use

### ISSUE THREAD

**Issue by dhalbert** [adafruit/circuitpython#1089]
`busio` (and presumably other base modules) do not do pin ownership, so it's possible to create multiple devices using the same pins, e.g.:
```
spi1 = busio.SPI(board.SCK,board.MOSI,board.MISO)
spi2 = busio.SPI(board.SCK,board.MOSI,board.MISO)
```
does not complain on nrf52. Or you could even create an SPI bus and and I2C bus on the same pins.

Symptoms noticed in #1088 (thanks @jerryneedell!)

**Comment by dhalbert** [comment]
Related to #1040

**Comment by dhalbert** [comment]
Fixed by #1040.
